app/presenters/queries/content_item_presenter.rb

```diff
@@ -1,182 +1,137 @@
 
-          most_recent_item["description"] = most_recent_item["description"]["value"] if most_recent_item["description"]
-
-          if output_fields.include?("internal_name")
-            details = most_recent_item.fetch("details")
-            most_recent_item["internal_name"] = details["internal_name"] || most_recent_item.fetch("title")
-          end
-
-          most_recent_item["publication_state"] = publication_state(draft, live)
-          most_recent_item["lock_version"] = most_recent_item.fetch("lock_version").to_i
-
-          if live
-            most_recent_item["live_version"] = live.fetch("lock_version").to_i
+            yielder.yield result
           end
-
-          most_recent_item.slice(*output_fields)
         end
       end
 
-      def join_supporting_objects(scope)
-        scope = State.join_content_items(scope)
-        scope = Translation.join_content_items(scope)
-        scope = Location.join_content_items(scope)
-        scope = UserFacingVersion.join_content_items(scope)
-        scope = LockVersion.join_content_items(scope)
-
-        scope
-      end
-
-      def select_fields(scope)
-        ordering_fields = order.keys.map(&:to_s)
-        scope.select(
-          *ContentItem::TOP_LEVEL_FIELDS,
-          "content_id",
-          "states.name as state_name",
-          "lock_versions.number as lock_version",
-          "translations.locale",
-          "locations.base_path",
-          *ordering_fields,
-        )
-      end
-
-      def output_fields
-        if fields
-          output_fields = fields
-        else
-          additional_fields = %w(
-            locale
-            base_path
-            lock_version
-            publication_state
-            live_version
-          )
-
-          output_fields = ContentItem::TOP_LEVEL_FIELDS + additional_fields
-        end
-
-        output_fields.map(&:to_s)
-      end
-
-      def publication_state(draft, live)
-        draft_lock_version = draft.fetch("lock_version") if draft
-        live_lock_version = live.fetch("lock_version") if live
-
-        if draft_lock_version && live_lock_version && (draft_lock_version > live_lock_version)
-          "redrafted"
-        elsif live
-          "live"
-        elsif draft
-          "draft"
-        else
-          raise "Something unexpected happened"
-        end
+      def parse_json_column(result, column)
+        return unless result.key?(column)
+        result[column] = JSON.parse(result[column])
       end
 
-      def detect_draft(items)
-        items.detect { |i| i.fetch("state_name") == "draft" }
+      def parse_int_column(result, column)
+        return unless result.key?(column)
+        result[column] = result[column].to_i
       end
 
-      def detect_live(items)
-        items.detect { |i| i.fetch("state_name") == "published" }
+      # It is substantially faster to evaluate directly against Postgres than
+      # to use: #as_json, #pluck or even ActiveRecord::Base.connection.execute
```

Comment:
Has this been demonstrated in a production environment?  Maybe an artefact of development environment.